Using "first" for indent can result in message about "NaN spaces" with nested arrays

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:** current master 
* **Node Version:** 7.2.1
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
config file is empty

<!-- Paste your configuration below: -->
```
```

**What did you do? Please include the actual source code causing the issue.**

Running command:

```sh
eslint --rule 'indent: [2, 2, {ArrayExpression: first}]' test.js
```

with contents of `test.js` as follows:

<!-- Paste the source code below: -->
```js
var foo = [
  [
     1, 2, 3,
     4, 5, 6
  ]
];
```

**What did you expect to happen?**

Either no error or else expected indentation of some positive integer.

**What actually happened? Please include the actual, raw output from ESLint.**

  2:3  error  Expected indentation of NaN spaces but found 2  indent
  5:3  error  Expected indentation of NaN spaces but found 2  indent

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -727,9 +727,19 @@ module.exports = {
                         if (parent.type === "VariableDeclarator" && parentVarNode.loc.start.line === parent.loc.start.line) {
                             nodeIndent = nodeIndent + (indentSize * options.VariableDeclarator[parentVarNode.parent.kind]);
                         } else if (parent.type === "ObjectExpression" || parent.type === "ArrayExpression") {
-                            nodeIndent += options[parent.type] * indentSize;
+                            if (typeof options[parent.type] === "number") {
+                                nodeIndent += options[parent.type] * indentSize;
+                            } else {
+                                const parentElements = node.parent.type === "ObjectExpression" ? node.parent.properties : node.parent.elements;
+
+                                nodeIndent = parentElements[0].loc.start.column;
+                            }
                         } else if (parent.type === "CallExpression" || parent.type === "NewExpression") {
-                            nodeIndent += (typeof options.CallExpression.arguments === "number" ? options.CallExpression.arguments : 1) * indentSize;
+                            if (typeof options.CallExpression.arguments === "number") {
+                                nodeIndent += options.CallExpression.arguments * indentSize;
+                            } else if (parent.arguments.indexOf(node) !== -1) {
+                                nodeIndent = parent.arguments[0].loc.start.column;
+                            }
                         } else if (parent.type === "LogicalExpression" || parent.type === "ArrowFunctionExpression") {
                             nodeIndent += indentSize;
                         }

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -1829,6 +1829,40 @@ ruleTester.run("indent", rule, {
         {
             code: "{\n}",
             options: [2, {ObjectExpression: 1}]
+        },
+        {
+            code:
+            "var foo = [\n" +
+            "  [\n" +
+            "    1\n" +
+            "  ]\n" +
+            "]",
+            options: [2, {ArrayExpression: "first"}]
+        },
+        {
+            code:
+            "var foo = [ 1,\n" +
+            "            [\n" +
+            "              2\n" +
+            "            ]\n" +
+            "];",
+            options: [2, {ArrayExpression: "first"}]
+        },
+        {
+            code:
+            "var foo = bar(1,\n" +
+            "              [ 2,\n" +
+            "                3\n" +
+            "              ]\n" +
+            ");",
+            options: [4, {ArrayExpression: "first", CallExpression: {arguments: "first"}}]
+        },
+        {
+            code:
+            "var foo =\n" +
+            "    [\n" +
+            "    ]()",
+            options: [4, {CallExpression: {arguments: "first"}, ArrayExpression: "first"}]
         }
     ],
     invalid: [

